frontend/src/components/productCard/ProductCard.tsx

```diff
@@ -57,33 +60,30 @@ export const ProductCard = ({ item }: ProductCardProps) => {
             </div>
             <div className={styles.avatarsAndUnits}>
                 <div className={styles.avatars}>
-                    {item.bidHistory.slice(-3).map((item) => {
-                        return (
-                            <div className={styles.avatar}>
-                                <ProfilePicture
-                                    url={item.bid.profileID.profilePicture}
-                                    width={"24px"}
-                                />
-                            </div>
-                        );
-                    })}
+                    {checkIsBidHistory() &&
```

Comment:
if you're sure that an empty history will be an empty array, you can skip the check here, since you can slice and map over an empty array